components/uniset/src/utils.rs

```diff
@@ -3,13 +3,16 @@ use std::{
     ops::{Bound::*, RangeBounds},
 };
 
-/// Returns whether the vector is sorted ascending non inclusive
-pub fn is_sorted(v: &[u32]) -> bool {
-    v.chunks(2).all(|chunk| chunk[0] < chunk[1])
+/// Returns whether the vector is sorted ascending non inclusive, of even length,
+/// and within the bounds of `0x0 -> 0x10FFFF` inclusive.
+pub fn is_valid(v: &[u32]) -> bool {
+    v.len() % 2 == 0
+        && v.chunks(2).all(|chunk| chunk[0] < chunk[1])
```

Comment:
You should be stricter.  The whole vector should be sorted, not just each non-overlapping pair.

Reply:
good catch, accidentally used `chunks` instead of [`windows`](https://doc.rust-lang.org/std/slice/struct.Windows.html).
Changed and added unit test to catch it.